src/llmcompressor/transformers/finetune/session_mixin.py

```diff
@@ -632,3 +643,54 @@ def _calculate_checkpoint_info(self, kwargs) -> Tuple[Optional[str], float]:
             ).epoch
 
         return checkpoint, epoch
+
+
+class OneshotSessionManagerMixIn:
```

Comment:
I dont think this class is needed. As the goal is to remove abstractions, we can initialize the session directly, this class isn't providing any additional.